cmd/algorand-indexer/main.go

```diff
@@ -97,6 +98,28 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(&cpuProfile, "cpuprofile", "", "", "file to record cpu profile to")
 	rootCmd.PersistentFlags().StringVarP(&pidFilePath, "pidfile", "", "", "file to write daemon's process id to")
 	rootCmd.PersistentFlags().BoolVarP(&doVersion, "version", "v", false, "print version and exit")
+
+	// Setup viper configuration file
+	viper.SetConfigName("indexer")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath(".")
+	viper.AddConfigPath("$HOME")
+	viper.AddConfigPath("$HOME/.indexer/")
+	viper.AddConfigPath("$HOME/.config/indexer/")
+	viper.AddConfigPath("/etc/indexer/")
```

Comment:
nit - I'd expect the name to have the word "config" in it. 

Also, let's make all of it constants - The list of config paths should just be a list

Reply:
maybe be like the binary name, "algorand-indexer" ?

Reply:
This is definitely a subjective one. I actually checked whether `indexer.config` was accepted before making the PR, which might have been nice. But it isn't.

I think having a yaml extension makes it pretty obvious that this is some sort of configuration. Since we don't have a specific data directory for indexer, I think that this implicit config file approach needs to be the service name. If we didn't, if there was a file named `config` in the home directory indexer would try to load it.

I don't feel strongly about any of these, but they are all reasonable options:
* indexer
* indexer-config
* algorand-indexer
* algorand-indexer-config (this one is maybe a bit much :laughing:)